### PR TITLE
Rewards panel now opens in 'First Run' state with Rewards page when wallet is corrupt

### PR DESCRIPTION
**Pull request by jasonrsadler**
Resolves https://github.com/brave/brave-browser/issues/3697

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:

1. Open Brave with this profile https://github.com/brave/brave-browser/files/2823589/Brave-Browser-Beta-05925.zip
2. Click Rewards panel and make sure that panel is at 'First Run' 'Join Now' screen.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc [hunk 1]
```diff
@@ -226,10 +226,14 @@ void LedgerImpl::OnLedgerStateLoaded(ledger::Result result,
...
-      SetConfirmationsWalletInfo(wallet_info);
-
-      LoadPublisherState(this);
-      bat_contribution_->OnStartUp();
+      if (GetPaymentId().empty() || GetWalletPassphrase().empty()) {
+        OnWalletInitialized(ledger::Result::CORRUPTED_WALLET);
+      } else {
+        SetConfirmationsWalletInfo(wallet_info);
+
+        LoadPublisherState(this);
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc]: even if wallet is corrupted, we need to load other stuff

Comment (jasonrsadler) [reply vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc]: In this particular case where panel and page won't interact with that data, we should still load? This section will still run if they create a new wallet on corrupt.

Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc]: if seed is corrupted for example user can still load everything and do recovery on rewards page. If we block it then user will be in strange state

Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc]: problem is that you now return two `OnWalletInitialized` calls if we load other stuff

Comment (jasonrsadler) [reply vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc]: I moved check to after publisher_state load

File: patches/extensions-browser-extension_event_histogram_value.h.patch [hunk 2]
```diff
@@ -20,6 +20,7 @@ index d29bd5d6776c488a5d24dc40ca102946044131f9..b953ecdf34dd83ffce42347785b38c38
 +  BRAVE_REWARDS_GET_NOTIFICATION,
 +  BRAVE_REWARDS_GET_ALL_NOTIFICATIONS,
 +  BRAVE_WALLET_FAILED,
++  BRAVE_WALLET_CORRUPTED,
```
Comment (NejcZdovc) [inline patches/extensions-browser-extension_event_histogram_value.h.patch]: patch is not needed as we use `BRAVE_START` for all of them

File: components/brave_rewards/browser/extension_rewards_service_observer.cc [hunk 3]
```diff
@@ -44,6 +44,12 @@ void ExtensionRewardsServiceObserver::OnWalletInitialized(
         extensions::api::brave_rewards::OnWalletCreated::kEventName,
         std::move(args)));
       event_router->BroadcastEvent(std::move(event));
+    } else if (result == ledger::Result::CORRUPTED_WALLET) {
```
Comment (NejcZdovc) [inline components/brave_rewards/browser/extension_rewards_service_observer.cc]: I know this was not introduced with this PR, but could we have only one `onWalletInitialized` and we would pass result in. This way we have only one function and handler for all of them. You could create separate issue for it and do it as follow up

File: patches/extensions-browser-extension_event_histogram_value.h.patch (hunk 2 above)
Comment (jasonrsadler) [reply patches/extensions-browser-extension_event_histogram_value.h.patch]: Removed and code adjusted.

File: components/brave_rewards/browser/extension_rewards_service_observer.cc (hunk 3 above)
Comment (jasonrsadler) [reply components/brave_rewards/browser/extension_rewards_service_observer.cc]: Created https://github.com/brave/brave-browser/issues/3974 to follow up

Comment (NejcZdovc) [reply components/brave_rewards/browser/extension_rewards_service_observer.cc]: thank you

File: components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts [hunk 4]
```diff
@@ -61,6 +62,10 @@ export const rewardsPanelReducer = (state: RewardsExtension.State | undefined, a
       state.walletCreating = false
       state.walletCreated = false
```
Comment (NejcZdovc) [inline components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts]: You need to set `state.walletCorrupted` to false here

Comment (NejcZdovc) [reply components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts]: same in `CREATE_WALLET`

Comment (jasonrsadler) [reply components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts]: Done.

**Comment by NejcZdovc** [comment]
blocked on https://github.com/brave/brave-core/pull/2122

**Pull request merged**